### PR TITLE
Add user auth

### PR DESCRIPTION
**Pull request by Eidenn005**
Add JwtUtil class
Add jwtt dependencies
Add UserRepository
Make changes in User and Roles classes according to needs
Add User dtos for login
Add Authentication service

### CODE REVIEW

**Review by ReamFOX — APPROVED** [review]
good

File: pom.xml [hunk 1]
```diff
@@ -18,7 +18,9 @@
         <lombok.mapstruct.binding.version>0.2.0</lombok.mapstruct.binding.version>
         <webmvc-ui.version>2.5.0</webmvc-ui.version>
         <maven.checkstyle.plugin.configLocation>checkstyle.xml</maven.checkstyle.plugin.configLocation>
+        <jjwt.version>0.12.5</jjwt.version>
         <telegrambot.version>6.9.7.1</telegrambot.version>
+        <jjwt.version>0.12.5</jjwt.version>
```
Comment (oksana-miazina) [inline pom.xml]: twice the same

File: pom.xml [hunk 2]
```diff
@@ -120,6 +138,22 @@
...
+            <groupId>io.jsonwebtoken</groupId>
+            <artifactId>jjwt-api</artifactId>
+            <version>${jjwt.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>io.jsonwebtoken</groupId>
+            <artifactId>jjwt-jackson</artifactId>
+            <version>${jjwt.version}</version>
+            <scope>runtime</scope>
+        </dependency>
```
Comment (oksana-miazina) [inline pom.xml]: Again duplication

File: src/main/java/ua/team3/carsharingservice/model/User.java [hunk 3]
```diff
@@ -56,4 +59,34 @@ public class User {
...
+    }
+
+    @Override
+    public boolean isAccountNonExpired() {
+        return UserDetails.super.isAccountNonExpired();
+    }
+
+    @Override
+    public boolean isAccountNonLocked() {
+        return UserDetails.super.isAccountNonLocked();
```
Comment (oksana-miazina) [inline src/main/java/ua/team3/carsharingservice/model/User.java]: Can we just return `true`?

File: src/main/java/ua/team3/carsharingservice/model/User.java [hunk 4]
```diff
@@ -56,4 +59,34 @@ public class User {
...
+    }
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return UserDetails.super.isCredentialsNonExpired();
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return UserDetails.super.isEnabled();
```
Comment (oksana-miazina) [inline src/main/java/ua/team3/carsharingservice/model/User.java]: I think it should be related to isDeleted

**Review by oksana-miazina — APPROVED** [review]

File: src/main/java/ua/team3/carsharingservice/model/User.java (hunk 3 above)
Comment (Eidenn005) [reply src/main/java/ua/team3/carsharingservice/model/User.java]: done

File: pom.xml (hunk 2 above)
Comment (Eidenn005) [reply pom.xml]: done

File: pom.xml (hunk 1 above)
Comment (Eidenn005) [reply pom.xml]: done

File: src/main/java/ua/team3/carsharingservice/model/User.java (hunk 4 above)
Comment (Eidenn005) [reply src/main/java/ua/team3/carsharingservice/model/User.java]: Its related to UserDetail.
Leave as is.

**Pull request merged**